### PR TITLE
Backport of Add ECS audit logging docs into release/1.12.x

### PR DESCRIPTION
**Pull request by hc-github-team-consul-core**
## Backport

This PR is auto-generated from #13729 to be assessed for backporting due to the inclusion of the label backport/1.12.

**WARNING** automatic cherry-pick of commits failed. Commits will require human attention.

The below text is copied from the body of the original PR.

---

### Description
Adding documentation for ACL audit logging for Consul on ECS. 

### PR Checklist

* [x] ~updated test coverage~
* [x] external facing docs updated
* [x] not a security concern

### CODE REVIEW

File: website/content/docs/ecs/terraform/secure-configuration.mdx [hunk 1]
```diff
@@ -136,8 +136,7 @@ resource "aws_secretsmanager_secret_version" "gossip_key" {
 
 ### Enable secure deployment
 
-Add the following configurations to enable secure deployment. The `acl_secret_name_prefix`
-should be the same as the `name_prefix` you provide to the ACL controller module.
+Add the following configurations to enable secure deployment.
```
Comment (trujillo-adam) [inline website/content/docs/ecs/terraform/secure-configuration.mdx:139]: I don't know that it is clear that the following configuration is an example/template. Maybe we should say something like "Add the configurations described in the [Secure configurations table] to secure your deployment. You can copy the following example and make adjustments as necessary."
We would need to add a header to the table to link to.

**Review by trujillo-adam — APPROVED** [review]
Good enough as is. I just added one suggestion/comment to consider.

File: website/content/docs/ecs/enterprise.mdx [hunk 2]
```diff
@@ -121,3 +121,29 @@ module "my_task" {
...
+
+### Audit Logging
+Consul on ECS supports [audit logging](/docs/enterprise/audit-logging) when Consul Enterprise clients are used.
+This feature has the following requirements:
+
+- ACLs must be enabled.
+- `mesh-task` must use a Consul Enterprise client image.
+- `gateway-task` must use a Consul Enterprise client image.
+
+The following example demonstrates how to create a `mesh-task` with audit logging enabled.
```
Comment (sanon-dev) [inline website/content/docs/ecs/enterprise.mdx:133]:
```suggestion
Consul on ECS supports [audit logging](/docs/enterprise/audit-logging) when using Consul Enterprise clients.
This feature has the following requirements:

- ACLs must be enabled.
- `mesh-task` must use a Consul Enterprise image.
- `gateway-task` must use a Consul Enterprise image.

To enable audit logging, set `audit_logging = true` when configuring the client.
```

Comment (sanon-dev) [reply website/content/docs/ecs/enterprise.mdx:133]: Adding this change to maintain consistency between main and this branch.

File: website/content/docs/ecs/manual/secure-configuration.mdx [hunk 3]
```diff
@@ -472,6 +472,45 @@ The following table describes the additional fields that must be included in the
 | [`acl.tokens.agent`](/docs/agent/config/config-files#acl_tokens_agent)          | string  | Consul client token which authorizes this agent with Consul servers.                    |
 | [`partition`](/docs/agent/config/config-files#partition-1)                      | string  | <EnterpriseAlert inline /> The Consul Enterprise admin partition this agent belongs to. |
 
+### Configure Audit Logging<EnterpriseAlert inline />
+[Audit logging](/docs/enterprise/audit-logging) is supported when ACLs are enabled and tasks are configured to use a Consul Enterprise client.
+To enable audit logging you must update the startup script to configure the Consul client config file with an `audit` stanza.
+
+The following example shows the additional content to include in the `consul-client` startup script to configure audit logs to be written to `stdout` of the `consul-client` container.
```
Comment (sanon-dev) [inline website/content/docs/ecs/manual/secure-configuration.mdx:479]:
```suggestion
[Audit logging](/docs/enterprise/audit-logging) is supported on clients running Consul Enterprise with ACLs enabled.
To enable audit logging, update the startup script to add an `audit` stanza to the Consul client configuration file.

The following example modifies the `consul-client` startup script to configure audit logs to be written to the `stdout` of the `consul-client` container.
```

Comment (sanon-dev) [reply website/content/docs/ecs/manual/secure-configuration.mdx:479]: Adding this change to maintain consistency between main and this branch.

File: website/content/docs/ecs/terraform/secure-configuration.mdx (hunk 1 above)
Comment (sanon-dev) [reply website/content/docs/ecs/terraform/secure-configuration.mdx:139]:
```suggestion
To enable secure deployment, add the following configuration to the task module.
```

Comment (sanon-dev) [reply website/content/docs/ecs/terraform/secure-configuration.mdx:139]: Adding this change to maintain consistency between main and this branch.

File: website/content/docs/ecs/terraform/secure-configuration.mdx [hunk 4]
```diff
@@ -166,6 +166,7 @@ The following table explains the `mesh-task` input variables relevant to a secur
 | `acls`                      | boolean  | If true, ACLs are enabled.                                                                                                                                                                                                                 |
 | `consul_http_addr`          | string   | The Consul server address. Required when `acls = true` in order to log in to Consul's AWS IAM auth method to obtain ACL tokens.                                                                                                            |
 | `consul_https_ca_cert_arn`  | string   | (optional) The Secrets Manager secret containing the CA cert for HTTPS communication with Consul servers. Required if the server's certificate is self-signed or signed by an internal CA. This is not required for Consul servers in HCP. |
+| `audit_logging`             | boolean  | <EnterpriseAlert inline /> (optional) If true, ACL audit logging is enabled.                                                                                                                                                               |
```
Comment (sanon-dev) [inline website/content/docs/ecs/terraform/secure-configuration.mdx:169]:
```suggestion
| `audit_logging` | boolean | <EnterpriseAlert inline /> (optional) If true, ACL audit logging is enabled. Consul client is configured to print audit logs to `stdout`. |
```

Comment (sanon-dev) [reply website/content/docs/ecs/terraform/secure-configuration.mdx:169]: Adding this change to maintain consistency between main and this branch.

**Review by sanon-dev — CHANGES_REQUESTED** [review]

**Review by sanon-dev — APPROVED** [review]

**Review by pglass — APPROVED** [review]

**Pull request merged**